### DISCUSSION TITLE
Stream using v2 API

### DISCUSSION THREAD

**Discussion by flxai** [tweepy/tweepy#2063] [Questions] [answered]
Hello and thanks for this project,

reading Tweepy's [docs on streaming](https://docs.tweepy.org/en/stable/streaming.html), it seems to me that this only allows for usage of Twitter's v1 API, since it requires the user to provide a consumer key, consumer secret, access token and access token secret. Is it possible or planned to be able to use Twitter's new v2 API to use filtered streaming in Tweepy?

**Comment by flxai** [comment]
Seems like Tweepy is already capable of this, using [`tweepy.StreamingClient.filter`](https://docs.tweepy.org/en/stable/streamingclient.html#tweepy.StreamingClient.filter):

> Filter and sample realtime Tweets with Twitter API v2

While usage appears to be self-explanatory, it seems that proper documentation is still lacking.

**Comment by Harmon758** [comment, marked as answer]
Tweepy has supported streaming with Twitter API v2, with `StreamingClient`, since [v4.6.0](https://github.com/tweepy/tweepy/releases/tag/v4.6.0), which was released almost a year ago.

Usage of `StreamingClient` is fully and specifically documented in [an entire section of the documentation page that you linked](https://tweepy.readthedocs.io/en/v4.12.1/streaming.html#using-streamingclient).

**Comment by flxai** [comment]
Thanks for pointing out these details @Harmon758. I think the documentation might benefit from adding which version of the API is used at [Using `Stream`](https://docs.tweepy.org/en/v4.12.1/streaming.html#using-stream) and [Using `StreamingClient`](https://docs.tweepy.org/en/v4.12.1/streaming.html#using-streamingclient). If I am not mistaken, `Stream` is using v1 and `StreamingClient` is using v2, correct? If so, I'd create a PR to make this more clear in the docs.

**Reply by Harmon758** [reply]
The very first section of the documentation on that page clearly explains this:
> [`Stream`](https://docs.tweepy.org/en/v4.12.1/stream.html#tweepy.Stream) allows [filtering](https://developer.twitter.com/en/docs/twitter-api/v1/tweets/filter-realtime/overview) and [sampling](https://developer.twitter.com/en/docs/twitter-api/v1/tweets/sample-realtime/overview) of realtime Tweets using Twitter API v1.1.
> 
> [`StreamingClient`](https://docs.tweepy.org/en/v4.12.1/streamingclient.html#tweepy.StreamingClient) allows [filtering](https://developer.twitter.com/en/docs/twitter-api/tweets/filtered-stream/introduction) and [sampling](https://developer.twitter.com/en/docs/twitter-api/tweets/volume-streams/introduction) of realtime Tweets using Twitter API v2.

**Reply by flxai** [reply]
Thanks for pointing it out. Maybe repeating it would be in order? Maybe I'm not the only one confused. But it's your decision.
